src/components/Property/RaiseWorkOrder/RaiseWorkOrderForm.js

```diff
@@ -67,30 +67,31 @@ const RaiseWorkOrderForm = ({
         ?.description
     }
 
-    if (priorityList.includes(description)) {
-      const existingCode = parseInt(
-        document.getElementById('priorityCode').value
-      )
+    if (getPriorityObjectByDescription(description)) {
       // Update priority when SOR code has priority attached if:
       // Priority description is blank, or there's only one sor code entry, or
       // when removing an SOR there's an existing entry with higher priority, or
       // the selected priority code is less than existing priority codes
       // (Higher priority as code gets lower)
+
       if (
-        !existingCode ||
+        !priorityCode ||
         rateScheduleItemsLength <= 1 ||
         existingHigherPriorityCode ||
-        code < existingCode
+        code < priorityCode
       ) {
-        if (errors?.priorityDescription) {
-          delete errors.priorityDescription
+        if (errors?.priorityCode) {
+          delete errors.priorityCode
         }
 
-        document.getElementById('priorityDescription').value = description
-
         setPriorityCode(
           getPriorityObjectByDescription(description)?.priorityCode
         )
+
+        setValue(
+          'priorityCode',
+          getPriorityObjectByDescription(description)?.priorityCode
+        )
```

Comment:
Without this, the priority code state update is not reflected in the select. I'm not sure right now if there's a better way - I'm sure there's a technique available in react-hook-form but I haven't found it yet.